frontend/src/metabase/browse/components/BrowseRedirect.tsx

```diff
@@ -0,0 +1,55 @@
+import { replace } from "react-router-redux";
+import { useEffect } from "react";
+import type { SearchResult } from "metabase-types/api";
+import { useSearchListQuery } from "metabase/common/hooks";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { getDefaultBrowseTab } from "../selectors";
+
+export const BrowseRedirect = () => {
+  const defaultTab = useSelector(getDefaultBrowseTab);
+
+  const shouldQueryModels = !defaultTab;
+
+  const modelsResult = useSearchListQuery<SearchResult>({
+    query: {
+      models: ["dataset"],
+      filter_items_in_personal_collection: "exclude",
+    },
+    enabled: shouldQueryModels,
+  });
+  const { data: models, error, isLoading } = modelsResult;
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    switch (defaultTab) {
+      case "models":
+        dispatch(replace("/browse/models"));
+        break;
+      case "databases":
+        dispatch(replace("/browse/databases"));
+        break;
+      default:
+        if (models) {
```

Comment:
I believe that if models is `undefined`, that means the model query is still loading, so `isLoading` will be true and the user will be shown the loading screen.

`if (models)` is kind of a code smell. I'll change it to `if (models !== undefined)` for explicitness.

I'm also not sure how we handle the loading state of the call to /api/settings on line 10

Reply:
> I believe that if models is undefined, that means the model query is still loading

This assumption isn't correct, you could have no data from a bad api response or an error occurred. I think it would be best to look at the isLoading and error values on `modelsResult` in addition to the data to make the correct choice here. 

As an aside, I believe this bug in our task board is based on the same assumption: https://github.com/metabase/metabase/issues/32793

Reply:
Yes, you're right. I think line 44 will catch this scenario?